samza-core/src/main/java/org/apache/samza/table/TableConfigGenerator.java

```diff
@@ -19,80 +19,24 @@
 
 package org.apache.samza.table;
 
-import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
-
 import org.apache.samza.config.Config;
-import org.apache.samza.config.JavaTableConfig;
-import org.apache.samza.table.descriptors.BaseTableDescriptor;
 import org.apache.samza.table.descriptors.TableDescriptor;
-import org.apache.samza.operators.TableImpl;
-import org.apache.samza.util.Util;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-
 /**
  * Helper class to generate table configs.
  */
 public class TableConfigGenerator {
   private static final Logger LOG = LoggerFactory.getLogger(TableConfigGenerator.class);
 
-  /**
-   * Generate table configurations given a list of table descriptors
-   * @param config the job configuration
-   * @param tableDescriptors the list of tableDescriptors
-   * @return configuration for the tables
-   */
-  static public Map<String, String> generateConfigsForTableDescs(Config config, List<TableDescriptor> tableDescriptors) {
-    return generateConfigsForTableSpecs(config, getTableSpecs(tableDescriptors));
-  }
-
-  /**
-   * Generate table configurations given a list of table specs
-   * @param config the job configuration
-   * @param tableSpecs the list of tableSpecs
-   * @return configuration for the tables
-   */
-  static public Map<String, String> generateConfigsForTableSpecs(Config config, List<TableSpec> tableSpecs) {
-    Map<String, String> tableConfigs = new HashMap<>();
-
-    tableSpecs.forEach(tableSpec -> {
-        // Add table provider factory config
-        tableConfigs.put(String.format(JavaTableConfig.TABLE_PROVIDER_FACTORY, tableSpec.getId()),
-            tableSpec.getTableProviderFactoryClassName());
-
-        // Generate additional configuration
-        TableProviderFactory tableProviderFactory =
-            Util.getObj(tableSpec.getTableProviderFactoryClassName(), TableProviderFactory.class);
-        TableProvider tableProvider = tableProviderFactory.getTableProvider(tableSpec);
-        tableConfigs.putAll(tableProvider.generateConfig(config, tableConfigs));
-      });
-
-    LOG.info("TableConfigGenerator has generated configs {}", tableConfigs);
-    return tableConfigs;
+  static public Map<String, String> generate(Config jobConfig, List<TableDescriptor> tableDescriptors) {
```

Comment:
Minor: From what I can tell, this is only used in one place (JobNodeConfigurationGenerator). Since this is fairly simple now, would prefer removing this class and inlining the generate method there.

Reply:
It is actually used in samba-frame, which is a separate MP.